common/utilities/string/string-utilities.h

```diff
@@ -0,0 +1,106 @@
+        inline bool string_to_value(const std::string& str, T& result)
+        {
+            try
+            {
+                size_t last_char_idx;
+                if (std::is_integral<T>::value)
+                {
+                        if (std::is_same<T, unsigned long>::value)
+                    {
+                        if (str[0] == '-') return false; //for explanation see 'NOTE for unsigned types'
+                        result = std::stoul(str, &last_char_idx);
+                    }
+                    else if (std::is_same<T, unsigned long long>::value)
+                    {
+                            if (str[0] == '-') return false;
+                        result = std::stoull(str, &last_char_idx);
+                    }
+                    else if (std::is_same<T, long>::value)
+                    {
+                        result = std::stol(str, &last_char_idx);
+                    }
+                    else if (std::is_same<T, long long>::value)
+                    {
+                        result = std::stoll(str, &last_char_idx);
+                    }
+                    else if (std::is_same<T, int>::value)
+                    {
+                        result = std::stoi(str, &last_char_idx);
+                    }
+                    else if (std::is_same<T, short>::value)
+                        { // no dedicated function in std
+                            result = std::stoi(str, &last_char_idx);
+                        }
+                    else if (std::is_same<T, unsigned short>::value)
+                        { // no dedicated function in std
+                            if (str[0] == '-') return false;
+                            result = std::stoul(str, &last_char_idx);
+                        }
+                    else if (std::is_same<T, unsigned int>::value)
+                        { // no dedicated function in std - unsgined in corresponds to 16 bit, and unsigned long corresponds to 32 bit
+                            if (str[0] == '-') return false; 
+                            result = std::stoul(str, &last_char_idx);
+                        }
+                    else
+                    {
+                            return false;
+                    }
+                }
+                else if (std::is_floating_point<T>::value)
+                {
+                    if (std::is_same<T, float>::value)
+                    {
+                        result = std::stof(str, &last_char_idx);
+                        if (std::isnan((float)result) || std::isinf((float)result)) return false; // if value is NaN or inf (or -inf), return false
+                    }
+                    else if (std::is_same<T, double>::value)
+                    {
+                        result = std::stod(str, &last_char_idx);
+                        if (std::isnan((double)result) || std::isinf((double)result)) return false;
```

Comment:
Make minor refinements:
1. (isnan || isinf) == !isfinite
2. Check for sub-normal value ( 0 < x < min_float) with `std::fpclassify`